iree/compiler/Codegen/Common/test/flatten_memref_subspan.mlir

```diff
@@ -344,3 +344,26 @@ func @load_store_rank_one_static_size_subspan_with_offset(%offset : index, %i: i
 //      CHECK:   %[[V:.+]] = memref.load %[[SPAN0]][%[[I0]]] : memref<?xf32>
 //      CHECK:   %[[I1:.+]] = affine.apply #[[MAP]]()[%[[I]], %[[OFFSET]]]
 //      CHECK:   memref.store %[[V]], %[[SPAN1]][%[[I1]]] : memref<?xf32>
+
+// -----
+
+func @projected_permutation_map(%arg0 : index, %arg1 : index)
```

Comment:
This is fully static and checks a specific form of dropping innermost size-1 dim. Can we add another test with dynamic shapes and real permutation maps?

Reply:
Ill drop that patch from the commit and see if all tests pass. Maybe this can be addressed as part of https://github.com/google/iree/issues/8580 .